CMakeLists.txt

```diff
@@ -535,19 +535,19 @@ if (CMAKE_COMPILER_IS_GNUCXX)
 endif()
 
 if(OPENDAQ_GENERATE_PYTHON_BINDINGS OR OPENDAQ_ENABLE_COVERAGE) #gcc coverage
-    set(PYTHON_VERSION 3.6)
-    set(COMPONENTS Interpreter)
+    set(OPENDAQ_PYTHON_VERSION_INTERNAL 3.8)
```

Comment:
Personally, I would:
* put variable renames inside this file into a separate commit
* pybind11 upgrade into a separate one
* and finally python version increase along with numpy dependency changes into its own commit